bundles/mapping/mapmodule/mapmodule.olcs.js

```diff
@@ -427,7 +430,81 @@ class MapModuleOlCesium extends MapModuleOl {
             });
         }
     }
+    /**
+     * Function to reset map move mode and 3D camera controls to default.
+     * This is needed since user might click reset map view on panbuttons when camera is in rotate mode.
+     */
+    _resetMoveMode () {
+        this.setCameraToMoveMode();
+        const cameraControlsPlugin = this._pluginInstances.CameraControls3dPlugin;
+        if (cameraControlsPlugin) {
+            cameraControlsPlugin.resetState();
+        }
+    }
+
+    setCameraToRotateMode () {
+        const camera = this.getCesiumScene().camera;
+        // Look at target in front at current direction
+        var distanceInMeters = 5;
+        camera.zoomIn(distanceInMeters);
+        var lookAtTarget = camera.position;
+        // Place camera to original position (x meters from look at target)
+        camera.constrainedAxis = Cesium.Cartesian3.UNIT_Z;
+        var offset = new Cesium.HeadingPitchRange(camera.heading, camera.pitch, distanceInMeters);
+        camera.lookAt(lookAtTarget, offset);
+        this._disableMapMoveControls();
+    }
+
+    setCameraToMoveMode () {
+        const camera = this.getCesiumScene().camera;
+        camera.constrainedAxis = undefined;
+        camera.lookAtTransform(Cesium.Matrix4.IDENTITY);
+        this._enableMapMoveControls();
+    }
+    _disableMapMoveControls () {
+        const styleClass = 'map-move-control-disabled';
+        const mapInMobileMode = Oskari.util.isMobile();
+        var controls;
 
+        if (mapInMobileMode) {
+            controls = ['.mobile-zoom-in', '.mobile-zoom-out', '.mobile-my-location', '.mobile-xy', '.mobile-north'];
+        } else {
+            controls = ['.maprotator', '.coordinatetool', '.mylocationplugin'];
+            // Zoom bar need more handling than other controls
+            const zoomBar = jQuery('.zoombar');
+            zoomBar.addClass(styleClass);
+            zoomBar.children().addClass(styleClass);
+            const slider = zoomBar.find('.slider');
+            slider.slider('disable');
+        }
+
+        controls.forEach(controlClass => {
+            const control = jQuery(controlClass);
+            control.addClass(styleClass);
+        });
+    }
+    _enableMapMoveControls () {
+        const styleClass = 'map-move-control-disabled';
+        const mapInMobileMode = Oskari.util.isMobile();
+        var controls;
+
+        if (mapInMobileMode) {
+            controls = ['.mobile-zoom-in', '.mobile-zoom-out', '.mobile-my-location', '.mobile-xy', '.mobile-north'];
+        } else {
+            controls = ['.maprotator', '.coordinatetool', '.mylocationplugin'];
+            // Zoom bar need more handling than other controls
+            const zoomBar = jQuery('.zoombar');
```

Comment:
This selector is so generic it might hit admin-tools etc as well

Reply:
Fixed